identity-iota/src/did/doc/document.rs

```diff
@@ -381,6 +382,19 @@ impl Document {
 
     Ok(DID::encode_key(message_id.encode_hex().as_bytes()))
   }
+
+  pub fn insert_service(&mut self, service: Service) -> bool {
+    if service.id().fragment().is_none() {
+      false
+    } else {
+      self.document.service_mut().append(service.serde_into().unwrap())
+    }
+  }
+
+  pub fn remove_service(&mut self, did: &DID) -> Result<()> {
+    self.document.service_mut().remove(did.deref());
```

Comment:
You can just do `*did` instead of calling `deref()` directly

Reply:
Its not possible -> &**did would work but its kinda a messy solution?